crates/tx5/src/backend.rs

```diff
@@ -0,0 +1,134 @@
+//! Backend modules usable by tx5.
+
+use std::io::Result;
+use std::sync::Arc;
+
+use futures::future::BoxFuture;
+
+use crate::{Config, PubKey};
+use tx5_core::deps::serde_json;
+
+#[cfg(feature = "backend-go-pion")]
+mod go_pion;
+
+/// Backend modules usable by tx5.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum BackendModule {
+    #[cfg(feature = "backend-go-pion")]
+    /// The Go Pion-based backend.
+    GoPion,
+
+    #[cfg(feature = "backend-webrtc-rs")]
+    /// The Webrtc-RS-based backend.
+    WebrtcRs,
+
+    /// The mock backend.
+    Mock,
+}
+
+impl Default for BackendModule {
+    #[allow(unreachable_code)]
+    fn default() -> Self {
+        #[cfg(feature = "backend-go-pion")]
```

Comment:
Are these already prevented from being enabled together?

Reply:
They are currently, but when we get to implementing it, I'd actually like them to *not* be exclusive, hence the fallback pattern in this default constructor. Not that we'd compile very often with both of them, but it would fit better into rust's features being additive paradigm.